### PR TITLE
Replace setInventoryKnown with a rolling bloom filter (rebase of #7100)

### PR DESCRIPTION
**Pull request by sipa**

### CODE REVIEW

**Comment by jtimon** [comment]
utACK

**Comment by gmaxwell** [comment]
ACK

File: src/net.cpp [hunk 1]
```diff
@@ -2342,7 +2342,7 @@ unsigned int SendBufferSize() { return 1000*GetArg("-maxsendbuffer", DEFAULT_MAX
 CNode::CNode(SOCKET hSocketIn, const CAddress& addrIn, const std::string& addrNameIn, bool fInboundIn) :
     ssSend(SER_NETWORK, INIT_PROTO_VERSION),
     addrKnown(5000, 0.001),
-    setInventoryKnown(SendBufferSize() / 1000)
+    filterInventoryKnown(50000, 0.000001)
```
Comment (dcousens) [inline src/net.cpp]: Why `50000` OOI?  Should this be a constant?

**Comment by dcousens** [comment]
utACK

**Comment by dcousens** [comment]
Also ACK on FP ratio of 0.000001:1

**Comment by petertodd** [comment]
utACK

**Comment by laanwj** [comment]
utACK

**Comment by morcos** [comment]
posthumous utACK.  nice.

**Pull request merged**